source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -207,3 +225,511 @@ void AMContinuousScanActionController::fillDataMaps(AMAgnosticDataAPIDataAvailab
+
+	interpolatedSize_ = (endAxisVaue-startAxisValue)/resolutionStep_;
+	qDebug() << "interpolatedSize is " << interpolatedSize_;
+
+	// Create the interpolated axis and midpoints
+	double currentAxisValue = startAxisValue;
+	interpolatedAxis_ = QVector<double>(interpolatedSize_);
+	interpolatedMidpoints_ = QVector<double>(interpolatedSize_);
+	for(int x = 0, size = interpolatedAxis_.count(); x < size; x++){
+		interpolatedAxis_[x] = currentAxisValue;
+		interpolatedMidpoints_[x] = currentAxisValue+(resolutionStep_/2);
+		currentAxisValue += resolutionStep_;
+	}
+
+	isUpScan_ = true;
+	if(axisFeedbackValues_.first() > axisFeedbackValues_.last())
+		isUpScan_ = false;
+
+
+	// Find the index mapping between interpolated points and feedback points
+	interpolatedMidpointsMappingIndices_ = QVector<double>(interpolatedSize_);
+	int currentInOrderEnergyLookupIndex = 0;
+	if(!isUpScan_)
+		currentInOrderEnergyLookupIndex = axisFeedbackValues_.count()-1;
+
+	double currentInOrderEnergyValue = axisFeedbackValues_.at(currentInOrderEnergyLookupIndex);
+	double lastInOrderEnergyValue = currentInOrderEnergyValue;
+	double fractionalIndex = 0;
+	double lastFractionalIndex = 0;
+
+	if(isUpScan_){
+		for(int x = 0, size = interpolatedSize_; x < size; x++){
+			double oneEnergyMidpoint = interpolatedMidpoints_.at(x);
+
+			while(oneEnergyMidpoint > currentInOrderEnergyValue){
+				currentInOrderEnergyLookupIndex++;
+				lastInOrderEnergyValue = currentInOrderEnergyValue;
+				currentInOrderEnergyValue = axisFeedbackValues_.at(currentInOrderEnergyLookupIndex);
+			}
+
+			lastFractionalIndex = fractionalIndex;
+			fractionalIndex = (currentInOrderEnergyLookupIndex-1) + ((oneEnergyMidpoint-lastInOrderEnergyValue)/(currentInOrderEnergyValue-lastInOrderEnergyValue));
+			interpolatedMidpointsMappingIndices_[x] = fractionalIndex;
+		}
+	}
+	else{
+		for(int x = 0, size = interpolatedSize_; x < size; x++){
+			double oneEnergyMidpoint = interpolatedMidpoints_.at(x);
+
+			while(oneEnergyMidpoint > currentInOrderEnergyValue){
+				currentInOrderEnergyLookupIndex--;
+				lastInOrderEnergyValue = currentInOrderEnergyValue;
+				currentInOrderEnergyValue = axisFeedbackValues_.at(currentInOrderEnergyLookupIndex);
+			}
+
+			lastFractionalIndex = fractionalIndex;
+			fractionalIndex = (currentInOrderEnergyLookupIndex+1) - ((oneEnergyMidpoint-lastInOrderEnergyValue)/(currentInOrderEnergyValue-lastInOrderEnergyValue));
+			interpolatedMidpointsMappingIndices_[x] = fractionalIndex;
+		}
+	}
+
+	return true;
+}
+
+bool AMContinuousScanActionController::generateInterpolatedScalerVectors()
+{
+	foreach(CLSAMDSScalerChannelDetector *scalerChannelDetector, scalerChannelDetectors_)
+		interpolatedScalerChannelVectors_.insert(scalerChannelDetector->name(), QVector<double>(interpolatedSize_));
+
+	// Loop over scaler interpolated vectors and fill them
+	QMap<QString, QVector<double> >::iterator b = interpolatedScalerChannelVectors_.begin();
```

Comment:
All the iterators used to be in the same function. Do any of the functions have more than one iterator now? I usually use "i" if there's only one, but the names could be better no doubt :O